test/distributed/_sharded_tensor/ops/test_linear.py

```diff
@@ -39,15 +45,15 @@ def _run_sharded_linear(self, spec, input_size, linear_size, sharded_dim):
         sharded_linear = torch.nn.Linear(*linear_size)
 
         # Copy the weights and bias from local linear
-        sharded_linear.weight = local_linear.weight
-        sharded_linear.bias = local_linear.bias
+        sharded_linear.weight = torch.nn.Parameter(local_linear.weight.detach().clone())
```

Comment:
It will raise a error complains about it. The error msg:

> TypeError: cannot assign 'torch.cuda.FloatTensor' as parameter 'weight' (torch.nn.Parameter or None expected)